### PR TITLE
[New] Add `html-closing-bracket-newline` (fixes #169)

### LINKED ISSUE #169: Rule Proposal `vue/closing-bracket-location`
## Please describe what the rule should do:

Provide a template rule similar to [`react/jsx-closing-bracket-location`](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-closing-bracket-location.md) for vue templates.

This rule would govern the indentation style for the closing bracket of html tags within vue components when a tag spans multiple lines. (single-line tags are fine)

## What category of rule is this? (place an "X" next to just one item)

[X] Enforces code style
[ ] Warns about a potential error
[ ] Suggests an alternate way of doing something
[ ] Other (please specify:)

#### Configuration

'vue/closing-bracket-location': ['tag-aligned' | 'line-aligned' | 'after-props' | 'props-aligned']

#### Examples:

The following would not produce errors:

```html
// 'vue/closing-bracket-location': [1, 'line-aligned']

<my-component
  really-long-param="foo"
  another-really-long-param="bar"
/>

Foo: <my-component
  really-long-param="foo"
  another-really-long-param="bar"
/>

<my-other-component
  really-long-param="foo"
  another-really-long-param="bar"
>
  Hello World!
</my-other-component>


// 'vue/closing-bracket-location': [1, 'tag-aligned']

<my-component
  really-long-param="foo"
  another-really-long-param="bar"
/>

Foo: <my-component
       really-long-param="foo"
       another-really-long-param="bar"
     />

<my-other-component
  really-long-param="foo"
  another-really-long-param="bar"
>
  Hello World!
</my-other-component>


// 'vue/closing-bracket-location': [1, 'after-props']

<my-component
  really-long-param="foo"
  another-really-long-param="bar" />

<my-other-component
  really-long-param="foo"
  another-really-long-param="bar">
  Hello World!
</my-other-component>


// 'vue/closing-bracket-location': [1, 'props-aligned']

<my-component
  really-long-param="foo"
  another-really-long-param="bar"
  />

<my-other-component
  really-long-param="foo"
  another-really-long-param="bar"
  >
  Hello World!
</my-other-component>

```

### PR DESCRIPTION
**Pull request by mysticatea**
This PR adds a new rule to enforce the location of closing brackets `>` of HTML tags.

### CODE REVIEW

**Comment by michalsnik** [comment]
I commented #169 as I'm having little doubts regarding this rule

**Comment by mysticatea** [comment]
I updated this PR to solve conflictions.

File: docs/rules/html-closing-bracket-newline.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+```json
+{
+    "html-closing-bracket-newline": ["error", {
+        "singleline": "never",
+        "multiline": "never"
+    }]
+}
+```
+
+- `singleline` ... the configuration for single-line elements. It's a single-line element if the element does not have attributes or the last attribute is on the same line of the opening bracket.
```
Comment (michalsnik) [inline docs/rules/html-closing-bracket-newline.md]: `as the opening bracket`

File: docs/rules/html-closing-bracket-newline.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    - `"never"` ... disallow line breaks before the closing bracket of elements. This is the default.
```
Comment (michalsnik) [inline docs/rules/html-closing-bracket-newline.md]: I'd remove `of elements` part

File: docs/rules/html-closing-bracket-newline.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    - `"always"` ... require one line break before the closing bracket of elements.
```
Comment (michalsnik) [inline docs/rules/html-closing-bracket-newline.md]: Here too

**Comment by mysticatea** [comment]
I updated this PR.

**Comment by michalsnik** [comment]
Great, thank you!

**Pull request merged**